heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -153,7 +163,45 @@ public PackingPlan pack() {
     Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
 
     Set<PackingPlan.ContainerPlan> containerPlans = PackingUtils.buildContainerPlans(
-        containerInstances, ramMap, this.defaultInstanceResources, paddingPercentage);
+        resourceCompliantRRAllocation.get(), ramMap, this.defaultInstanceResources,
+        paddingPercentage);
+    return new PackingPlan(topology.getId(), containerPlans);
+  }
+
+  /**
+   * Get a new packing plan given an existing packing plan and component-level changes.
+   *
+   * @return new packing plan
+   */
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
+    int adjustments = 0;
+    this.numAdjustments = 0;
+    this.numContainers = currentPackingPlan.getContainers().size();
+    this.containerId = 1;
+
+    int additionalContainers = computeNumAdditionalContainers(componentChanges, currentPackingPlan);
+    increaseNumContainers(additionalContainers);
```

Comment:
`increaseNumContainers` takes a delta to increase by not an absolute, so I was proposing two methods:
```
increaseNumContainers(delta)
resetNumContainers(absolute)
```
it's just about being consistent with our use of encapsulation. When we want to add containers we call a method to increase, which handles the operation. When we want to reset to a value though, we instead manipulate the values directly in multiple places. we'd have better encapsulation and consistency if we did both operations through methods that handles modifying of the underlying values.